src/bentoml/_internal/io_descriptors/multipart.py

```diff
@@ -238,13 +238,23 @@ async def from_http_request(self, request: Request) -> dict[str, t.Any]:
                 f"{self.__class__.__name__} only accepts `multipart/form-data` as Content-Type header, got {ctype} instead."
             ) from None
 
-        to_populate = zip(
-            self._inputs.values(), (await populate_multipart_requests(request)).values()
-        )
-        reqs = await asyncio.gather(
-            *tuple(io_.from_http_request(req) for io_, req in to_populate)
-        )
-        return dict(zip(self._inputs, reqs))
+        form_values = await populate_multipart_requests(request)
+
+        res = {}
+        for field, descriptor in self._inputs.items():
+            if field not in form_values:
+                break
+            res[field] = descriptor.from_http_request(form_values[field])
+        else:
```

Comment:
I think a note for `for..else` loop would be nice for the reader
```suggestion
        else:  # NOTE: This is similar to goto, when there is no break.
```

Reply:
we can also link https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops this as well for reference for the unheard.